src/Catalyst.Core.Modules.Kvm/DeltaExecutor.cs

```diff
@@ -171,19 +161,33 @@ private void Execute(Delta delta, ITxTracer txTracer, bool readOnly)
             // revert state if any fails (take snapshot)
             foreach (var publicEntry in delta.PublicEntries)
             {
-                Execute(publicEntry, stateUpdate, txTracer, readOnly);
+                Execute(publicEntry, stateUpdate, txTracer);
             }
-
+            
+            var spec = _specProvider.GetSpec(stateUpdate.Number);
+            _storageProvider.Commit(txTracer.IsTracingState ? txTracer : null);
+            _stateProvider.Commit(spec, txTracer.IsTracingState ? txTracer : null);
+            
             if (!readOnly)
             {
-                // we should assign block rewards here (or in Ledger)
-                _stateProvider.CommitTree();
+                // IMPORTANT: if this line is removed then the trie state root will not be recalculated before saving in the DB
+                // bad design - to change
+                Keccak newStateRoot = _stateProvider.StateRoot;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'newStateRoot'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4979613)